### PR TITLE
Use modify instead of write in the uart configuration.

### PR DESCRIPTION
**Pull request by pierre-rouanet**
Replace all *write* occurrences with the *modify* version.

**Should be thoroughly check before merged!**

Other occurrences can be found in gpio, ad, etc. Tracked in #24.

### CODE REVIEW

File: stm32f0_hal/src/uart.rs [hunk 1]
```diff
@@ -131,11 +131,11 @@ impl Uart {
                             .disabled()
                     });
                     // Configure UART : baudrate
-                    uart1.brr.write(|w| {
+                    uart1.brr.modify(|_, w| {
```
Comment (nicolas-rabault) [inline stm32f0_hal/src/uart.rs]: I think those ones (baudrates registers) shouldn't be modify but write, @csagonero ?

File: stm32f0_hal/src/uart.rs [hunk 2]
```diff
@@ -131,11 +131,11 @@ impl Uart {
...
                         w.div_fraction()
                             .bits((FREQUENCY / (baudrate / 2)) as u8 & 0x0F)
                     });
-                    uart1.brr.write(|w| {
+                    uart1.brr.modify(|_, w| {
```
Comment (nicolas-rabault) [inline stm32f0_hal/src/uart.rs]: here is the second one

File: stm32f0_hal/src/uart.rs [hunk 3]
```diff
@@ -238,11 +238,11 @@ impl Uart {
                             .disabled()
                     });
                     // Configure UART : baudrate
-                    uart3.brr.write(|w| {
+                    uart3.brr.modify(|_, w| {
```
Comment (nicolas-rabault) [inline stm32f0_hal/src/uart.rs]: Same thing here.

File: stm32f0_hal/src/uart.rs (hunk 1 above)
Comment (csagonero) [reply stm32f0_hal/src/uart.rs]: Must be write ...

File: stm32f0_hal/src/uart.rs (hunk 2 above)
Comment (csagonero) [reply stm32f0_hal/src/uart.rs]: same

File: stm32f0_hal/src/uart.rs (hunk 3 above)
Comment (csagonero) [reply stm32f0_hal/src/uart.rs]: again

File: stm32f0_hal/src/uart.rs [hunk 4]
```diff
@@ -238,11 +238,11 @@ impl Uart {
...
                         w.div_fraction()
                             .bits((FREQUENCY / (baudrate / 2)) as u8 & 0x0F)
                     });
-                    uart3.brr.write(|w| {
+                    uart3.brr.modify(|_, w| {
```
Comment (csagonero) [inline stm32f0_hal/src/uart.rs]: baudrate must be write.

**Review by nicolas-rabault — APPROVED** [review]

**Review by csagonero — APPROVED** [review]

**Pull request merged**